samcli/hook_packages/terraform/hooks.py

```diff
@@ -0,0 +1,344 @@
+"""
+Terraform hooks implementation
+"""
+
+from dataclasses import dataclass
+import json
+import os
+from subprocess import run, CalledProcessError
+from tempfile import NamedTemporaryFile
+from typing import Any, Callable, Dict
+import hashlib
+from samcli.lib.hook.exceptions import PrepareHookException
+from samcli.lib.hook.hooks import Hooks
+
+from samcli.lib.utils.hash import str_checksum
+from samcli.lib.utils.resources import (
+    AWS_LAMBDA_FUNCTION as CFN_AWS_LAMBDA_FUNCTION,
+)
+
+TF_AWS_LAMBDA_FUNCTION = "aws_lambda_function"
+SUPPORTED_RESOURCE_TYPES = [TF_AWS_LAMBDA_FUNCTION]
+PROVIDER_NAME = "registry.terraform.io/hashicorp/aws"
+
+# max logical id len is 255
```

Comment:
not sure if this is true, got it from CDK (where the logic for building the logical ID was partially inspired from): https://github.com/aws/aws-cdk/blob/3a7975c35aed53b94d0ea5577e41ad70a81e1390/packages/%40aws-cdk/core/lib/private/uniqueid.ts#L21

Reply:
I believe the limitation comes from CFN